pages/api/search/leaderboard/points/index.ts

```diff
@@ -0,0 +1,65 @@
+import { withCors } from "middleware";
+import { NextApiRequest, NextApiResponse } from "next";
+import { WhereOptions } from "sequelize";
+
+import models from "db/models";
+
+import { calculateLeaderboardScore } from "helpers/leaderboard-score";
+import paginate, { calculateTotalPages } from "helpers/paginate";
+
+async function get(req: NextApiRequest, res: NextApiResponse) {
+  try {
+    const whereCondition: WhereOptions = {};
+
+    const { address, page, order, sortBy } = req.query || {};
+
+    if (address) whereCondition.address = address;
+    
+    const leaderboard = await models.leaderBoard.findAndCountAll(paginate({
+      attributes: {
+        exclude: ["id"],
+      },
+      where: whereCondition,
+      nest: true,
+      raw: true
+    }, req.query, [[sortBy || "numberNfts", order || "DESC"]]))
+      .then(async (items) => {
+        return Promise.all(items.rows.map(async (item) => {
```

Comment:
if all you need is the addresses of the users, you can join them all and then make one big request with
```ts
model.user.findAll({where: { address: { [Op.in]: items.rows.map(({address}) => address.toLowerCase()) } }})
```

You can easily separate the `leaderboard` constant and this _then_ into its own constants; this way you can then access the leaderboard prop to reduce/map the result found on the batch-user findAll.